Release 3.3.0 (#5652)
open-formulieren/open-forms: CHANGELOG.rst

   on your data and available resources for the database.

.. warning::

    In this release, we reworked the internal data type information. To ensure submitted
    data of existing submissions is formatted correctly, submitted variables need to be
    processed. Note that we include this script instead of a data migration, so it can be
    run separately, as it can take up to an hour to complete the entire operation for large
    environments.

    .. code-block:: bash

        # in the container via ``docker exec`` or ``kubectl exec``:
        python /app/bin/fix_submission_value_variable_missing_fields.py

.. warning::

    For the email and confirmation templates, and the registration backends, we changed
    the way that the data is generated. In case of key conflicts between static, component,
    and user-defined variables, the static variables will take precedence. Previously, the
    component and user-defined variables would override the static variables. Our validation
    guards against the use of keys that are already present in the static variables, but this
    does not cover old forms and newly-added static variables.

> 

Reviewer: This is missing the "Major features" section, see the other releases likes 3.2.0 and 3.1.0.

```suggestion
Major features
--------------

**💳 Worldline (Ogone replacement) support**

...

**🗺️ Map layers and advanced interactions**

...

**🚸 Children component with prefill**

...

**📈 Added application metrics**

...
```

Detailed changes
----------------

**New features**

> * [:backend:`5618`] Add support for WMS on map image in submission pdf.

Reviewer: I'd have a single entry referring to the maps epic and then nested bullets inside it for the individual aspects.

* [:backend:`5268`] The registration plugins now support the ``children`` component type.
* [:backend:`5095`] When authenticated via OpenID Connect (DigiD, eHerkenning, organization), upon
  submission completion you are now logged out at the identity provider.
* [:backend:`4951`] The map component in the summary PDF is now an image instead of the textual location
  coordinates.
* [:backend:`5269`] You can now use the ``children`` component data as source data for a
  repeating group to provide additional information with the new "synchronize variables" logic

>   action type.

Reviewer: I'd also group those family members' changelog entries together for a more coherent changelog.

* [:backend:`5575`] Added support for WMS tile layers in the map component, along
  with import/export functionality.
* [:backend:`5479`] The eIDAS (via OIDC) configuration is now simplified - you can specify
  which claims can contain a BSN and/or Pseudo ID.
* [:backend:`5060`] Redis Sentinel is now supported as high availability strategy for the background
  jobs message broker.
* [:backend:`5419`] Added prefill plugins for the new Yivi and eIDAS authentication backends.
* [:backend:`5515`] Yivi Attribute groups now have a system-generated unique identifier.
* [:backend:`5515`] You can now export and import Yivi attribute groups.
* [:backend:`5253`] The BRT (grey, pastel, water) background tile layers are now available by
  default in an Open Forms instance.
* [:backend:`5251`] The identifier field from the map background tile layer is now automatically
  populated from its label.

> * [:backend:`4879`] Finished the support for Worldline as payment provider:

Reviewer: ```suggestion
* [:backend:`4879`] Added support for Worldline as payment provider:
```
the stable version changelog should be logical for someone not having read the alpha release notes  - that's why we copy over those entries :)

    - Support for Worldline's ``variant`` and ``descriptor`` fields.
    - The merchant reference is generated by Open Forms, similar to the Ogone plugin.
    - Ogone merchants are automatically migrated where possible.
    - Webhook configuration (if configured in a older patch release) is automatically migrated.
    - Moved the location in the admin of the feedback URL's used for the Worldline
      payment webhooks to the webhook configuration page.
    - Added a bulk action to migrate forms from Ogone to Wordline.

* [:backend:`5133`] Added a feature flag to enable the new *experimental* renderer.
* [:backend:`3999`] Added support for Open Telemetry metrics. All available metrics and details can be
  found in the "Observability" documentation.

> * [:backend:`5478`] Added additional Yivi documentation.

Reviewer: we should probably group everything about Yivi and eIDAS together for better release notes structure

* [:backend:`5451`, :backend:`3999`] Added Open Telemetry support. The following metrics
  are now exposed:

    - HTTP request durations.
    - Number of "active requests".
    - Number of users, logins, logouts, failed logins, axes lockouts.
    - Submission starts, completions, paused and steps submitted.
    - Total amount of submissions by form or lifecycle stage.

>     - Attachment file sizes and amount uploaded.

Reviewer: this is obsoleted by the entry two lines above which refers to the Observability documentation. Just included the additional github issue reference is sufficient.

* [:backend:`2324`] Reworked part of the logic engine in preparation of further
  performance improvements, so that we can correctly reason about variable data types.

> * [:backend:`4879`] You can now use the "Worldline" payment provider, which is the

Reviewer: should be incorporated in the group about worldline that's mentioned earlier

  successor to Ogone legacy.

> * [:sdk:`825`] Added children component and updated email digest.

Reviewer: should be group together with the related entries

* [:backend:`5382`] Forms now have an internal remarks field.
* [:backend:`5268`] Added "Partners Roltype" and "Partners omschrijving" registration
  configuration options for the ZGW APIs and StUF-ZDS registration plugins.
* [:backend:`5428`] Updated the eIDAS (OIDC) LoA-Levels.

**Bugfixes**

> * [:backend:`5646`] Updated the children component styling.

Reviewer: no need to mention "bug fixes" for features that haven't been released in a stable version - it creates unnecessary panic

> * [:backend:`5646`] Fixed children and partner information not showing properly in
>   the summary PDF.
> * Added a missing closing tag for the children preview.

Reviewer: ```suggestion
```
no point in listing these - the feature hadn't been released in its broken form

* [:backend:`5615`] Fixed ZGW registrations which also contained zaakeigenschappen.
* [:backend:`5507`] Fixed mimetype detection for .msg files.

> * Fix sync variables resetting modified items.

Reviewer: hasn't been released

```suggestion
```

* [:backend:`5384`] Fixed form export references to Objects API groups which can be
  provisioned through setup-configuration.
* [:backend:`5527`] Fixed all step data being returned during logic check of saved
  submission step instead of only the data that has changed.
* [:backend:`5475`] Fixed Yivi claims with periods not being usable in logic.
* [:backend:`5271`] Fixed false positives being reported in the digest email when you
  have logic rules that use the ``reduce`` operation.
* [:backend:`5481`] Fixed user defined variables not being queried based on the form of
  the current submission.
* [:backend:`5471`] Fixed BRP "doelbinding" advanced options not becoming available
  when using family members components.
* [:backend:`5340`] Fixed error handling during the validation of registration backends.
* [:backend:`5454`] Fixed Piwik Pro debug mode no longer working.
* [:backend:`5413`] Fixed uploading filenames with soft-hyphens not passing form validation.
* Fixed a crash when rendering e-mail HTML where links (anchor tags) contain bold or
  italic formatting elements.

**Project maintenance**

* Optimized the migration which has the potential to run longer than usual.

> * Added a progressbar for the migration which has the potential to run longer than usual.

Reviewer: ```suggestion
* Added a progressbar to the data backfill upgrade script.
```
